sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -255,13 +256,20 @@ public void close() {
      * (LockDuration). If processing of the message requires longer than this duration, the lock needs to be renewed.
      * For each renewal, the lock is reset to the entity's LockDuration value.
      *
-     * @param message to be used.
+     * @param receivedMessage to be used.
      *
      * @return The {@link Mono} the finishes this operation on service bus resource.
      */
-    public Instant renewMessageLock(ServiceBusReceivedMessage message) {
-        //TODO(feature-to-implement)
-        return null;
+    public Mono<Instant> renewMessageLock(ServiceBusReceivedMessage receivedMessage) {
+        return connectionProcessor
+            .flatMap(connection -> connection.getManagementNode(entityPath))
+            .flatMap(serviceBusManagementNode -> {
+                return serviceBusManagementNode
+                    .renewMessageLock(receivedMessage)
+                    .doOnNext(instant -> {
```

Comment:
Do on next should be for side effects like logging. To force it as part of the pipeline, you can use map.